### DISCUSSION TITLE
The strategy doesn't execute

### DISCUSSION THREAD

**Discussion by analyticray** [kernc/backtesting.py#574] [Q&A]
I have sample data as follows:
![image](https://user-images.githubusercontent.com/43358650/150033379-fd4b0e08-df2f-4970-b8af-3a899062461c.png)

I would like to execute the buy when StartTrade is > 0 and close the position (sell) when endTrade >0. However, the strategy doesn't execute all (no trades) when I execute the following code:

class SmaCross(Strategy):

    def init(self):
        # Precompute the two moving averages
        self.starTrade = self.data.starTrade 
        self.EndTrade = self.data.EndTrade 
    
    def next(self): 
        if (self.starTrade[-1] > 0): 
            self.position.close()
            self.buy()

        elif (self.position and self.EndTrade[-1] > 0): 
            self.position.close()
            self.sell()

Can you please advise on what went wrong? 

Thanks.

**Comment by kernc** [comment]
If you `print(self.starTrade)`, you'll see its length isn't incremented throughout the run, rather it stays the initial size. However, `print(self.data.starTrade)` _would_ show incremented in `next()` since [`Strategy.data`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data) is.

Therefore, either of these should work:
```py
    def init(self):
        # Make it an indicator. Indicators' lengths are auto-incremented
        self.starTrade = self.I(lambda: self.data.starTrade)
```
or:
```py
    def next(self):
        # Strategy.data is auto-incremented
        if self.data.starTrade[-1] > 0:
            ...
```
And the same for the other one.
